packages/ag-charts-community/src/chart/chart.ts

```diff
@@ -930,14 +941,18 @@ export abstract class Chart extends Observable implements AgChartInstance {
     }
 
     async processData() {
+        if (typeof this.data === 'function') {
+            this.data = await this.dataLazyLoader.full(this.data);
+        }
+
```

Comment:
I would perhaps abstract this away or always invoke it (and have a no-op behaviour if `data` is a static array)?